pypxe/server.py

```diff
@@ -161,15 +166,14 @@ def main():
             args = parse_cli_arguments() # re-parse, CLI options take precedence
 
         # warn the user that they are starting PyPXE as non-root user
-        if os.getuid() != 0:
-            print >> sys.stderr, '\nWARNING: Not root. Servers will probably fail to bind.\n'
-
+        if os.geteuid() != 0:
+            sys.exit("You need to have root privileges to run this script.\nPlease try again, this time using 'sudo'. Exiting.")
```

Comment:
We don't necessarily need to exit here, if the user is only trying to run services with non privileged ports they should still be allowed to continue, but with a warning.

In fact, more logic might be appreciated, and we only print/exit if any of the ports to be used are <1024 and we are non root